sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -88,14 +86,14 @@ object DateTimeUtils {
    * Returns the number of days since epoch from java.sql.Date.
    */
   def fromJavaDate(date: Date): SQLDate = {
-    millisToDays(date.getTime)
+    microsToDays(Math.multiplyExact(date.getTime, MICROS_PER_MILLIS))
```

Comment:
shall we call `millisToMicros(date.getTime)` instead of `Math.multiplyExact(date.getTime, MICROS_PER_MILLIS)`?